examples/src/bin/msgraph.rs

```diff
@@ -0,0 +1,103 @@
+//! Showcases how to define and use a nonstandard token type.
+//!
+//! Note: MSGraph requires you to set `client_id` and `client_secret` as extra
+//! parameters when performing the token exchange (see below).
+
+use oauth2::{Client, Url, StandardToken};
+
+use anyhow::{anyhow, Result};
+
+
+
+pub struct ConfigMS {
+    pub client_id: String,
+    pub client_secret: String,
+    pub tenant_domain: String,
+}
+
+pub fn config_from_args_ms(name: &str) -> Result<ConfigMS> {
+    let app = clap::App::new(name)
+        .about("Testing out OAuth 2.0 flows")
+        .arg(
+            clap::Arg::with_name("client-id")
+                .takes_value(true)
+                .long("client-id")
+                .help("Client ID to use."),
+        )
+        .arg(
+            clap::Arg::with_name("client-secret")
+                .takes_value(true)
+                .long("client-secret")
+                .help("Client Secret to use."),
+        )
+        .arg(
+            clap::Arg::with_name("tenant-domain")
+                .takes_value(true)
+                .long("tenant-domain")
+                .help("Tenant domain to use."),
+        );
+
+    let m = app.get_matches();
+
+    let client_id = m
+        .value_of("client-id")
+        .ok_or_else(|| anyhow!("missing: --client-id <argument>"))?
+        .to_string();
+    let client_secret = m
+        .value_of("client-secret")
+        .ok_or_else(|| anyhow!("missing: --client-secret <argument>"))?
+        .to_string();
+
+    let  tenant_domain = m
+        .value_of("tenant-domain")
+        .ok_or_else(|| anyhow!("missing: --tenant-domain <argument>"))?
+        .to_string();
+
+    Ok(ConfigMS {
+        client_id,
+        client_secret,
+        tenant_domain,
+    })
+}
+
+/*
+Auth OAuth { 
+	access_token: None, 
+	scopes: {"User.ReadAll", "https://graph.microsoft.com/.default"}, 
+	credentials: 
+	{
+		"access_token_url": "https://login.microsoftonline.com/{tenant-domain}/oauth2/token", 
+		"authorization_url":"https://login.microsoftonline.com/{tenant-domain}/oauth2/authorize", 
+		"client_id": 		"337b2164-0006-41fc-a243-710f3d63c889", 
+		"client_secret": 	"6_uEpej8_vG.-Ph33Pg_Iv82u6QnP~30-E", 
```

Comment:
Accidentally included?

Reply:
:exploding_head: Doh... Always the little things... 
I noticed that after checking in and removed and regenerated the secret.

Reply:
I have reformatted and checked in for review